### PR TITLE
fix(api): reimplement Regex to replace shortcode.

### PR DESCRIPTION
**Pull request by jstirnaman**
- Update README with more detailed instructions for updating API docs.
- Use regular expression instead of string to replace URL shortcode, just in case of extra whitespace or line breaks.
- Look for shortcode in additional nodes, especially Parameters.
- Regenerate docs with URL shortcode replacement.

- [ ] Signed the [InfluxData CLA](https://www.influxdata.com/legal/cla/)
  ([if necessary](https://github.com/influxdata/docs-v2/blob/master/CONTRIBUTING.md#sign-the-influxdata-cla))
- [ ] Rebased/mergeable

### CODE REVIEW

File: api-docs/README.md [hunk 1]
```diff
@@ -1,4 +1,172 @@
...
+   ```sh
+   git checkout docs-release/influxdb-oss
+   git rebase -i OSS_RELEASE_TAG
+   git push -f origin docs-release/influxdb-oss
+   ```
+
+   Replace the following:
+
+   - **`OSS_RELEASE_TAG`**: release tag name or SHA from **Step 3**.
+
```
Comment (kelseiv) [inline api-docs/README.md:75]:
```suggestion
5. Enter the following commands into your terminal to update `docs-release/influxdb-oss` branch to the OSS release commit and rebase the branch to the [latest release of InfluxDB OSS](#how-to-find-the-api-spec-used-by-an-influxdb-oss-version), replacing **`OSS_RELEASE_TAG`** with the SHA from step 3.

   ```sh
   git checkout docs-release/influxdb-oss
   git rebase -i OSS_RELEASE_TAG
   git push -f origin docs-release/influxdb-oss
   ```
```

File: api-docs/README.md [hunk 2]
```diff
@@ -1,4 +1,172 @@
...
+   ```
+
+   Replace the following:
+
+   - **`OSS_RELEASE_TAG`**: release tag name or SHA from **Step 3**.
+  
+6. Go into your `docs-v2` directory and create a branch for your changes--for example:
+
+   ```sh
+   cd ~/github/docs-v2
```
Comment (kelseiv) [inline api-docs/README.md:79]:
```suggestion
   cd ~/docs-v2
```

File: api-docs/README.md [hunk 3]
```diff
@@ -1,4 +1,172 @@
...
+
+   ```sh
+   # In your terminal, go to the `docs-v2/api-docs` directory:
+   cd ./api-docs
+
+   # Create the directory:
+   mkdir v2.3
+   ```
+
+8. Enter the following commands into your terminal to fetch and process the contracts:
```
Comment (kelseiv) [inline api-docs/README.md:94]:
```suggestion
8. In `mkdir v2.3`, enter the following commands into your terminal to fetch and process the contracts:
```

File: api-docs/README.md [hunk 4]
```diff
@@ -1,4 +1,172 @@
...
+
+   ```sh
+   # Fetch the contracts, apply customizations, and bundle.
+   sh getswagger.sh oss
+   ```
+
+   To troubleshoot errors, see [how to generate API docs locally](#how-to-generate-api-docs-locally).
```
Comment (kelseiv) [inline api-docs/README.md:101]:
```suggestion
```

File: api-docs/README.md [hunk 5]
```diff
@@ -1,4 +1,172 @@
...
+
+   ```sh
+   cd ~/github/openapi
+   ```
+
+2. Enter the following commands into your terminal to checkout `docs-release/influxdb-oss` branch:
+
+   ```sh
+   git fetch -ap
+   git checkout -t docs-release/influxdb-oss
```
Comment (kelseiv) [inline api-docs/README.md:125]: does it have to be `docs-release`?

File: api-docs/README.md [hunk 6]
```diff
@@ -1,4 +1,172 @@
...
+3. Cherry-pick the new documentation commits into the release branch and push to the remote branch--for example:
+
+   ```sh
+   git cherry-pick [COMMIT_SHAs]
+   git push -f origin docs-release/influxdb-oss
+   ```
+
+   Replace the following:
+
+   - **`[COMMIT_SHAs]`**: one or more commit SHAs (space-separated).
```
Comment (kelseiv) [inline api-docs/README.md:137]:
```suggestion
3. Cherry-pick the commits with the updated description fields, and push the commits to the remote branch, replacing **`[COMMIT_SHAs]`** (one or more commit SHAs (space-separated))--for example:

   ```sh
   git cherry-pick [COMMIT_SHAs]
   git push -f origin docs-release/influxdb-oss
   ```
```

File: api-docs/README.md [hunk 7]
```diff
@@ -1,4 +1,172 @@
...
+
+7. Enter the following commands into your terminal to generate the API docs with Redocly:
+
+   ```sh
+   sh generate-api-docs.sh
+   ```
+
+8. Commit the changes, push your branch to `influxdata/docs-v2`, and create a PR against the `master` branch.
+
+## How to generate InfluxDB API docs
```
Comment (kelseiv) [inline api-docs/README.md:169]:
```suggestion
## Generate InfluxDB API docs
```

File: api-docs/README.md [hunk 8]
```diff
@@ -30,9 +201,10 @@ api-docs/
 
 ### InfluxDB Cloud version
 
-Because InfluxDB Cloud releases are frequent, we make no effort to version the
-Cloud API spec. We regenerate API reference docs from `influxdata/openapi`
-**master** as features are released.
+Because InfluxDB Cloud releases are frequent and not versioned, we make no effort to version the
+Cloud API spec.
```
Comment (kelseiv) [inline api-docs/README.md:205]:
```suggestion
InfluxDB Cloud releases are frequent and not versioned, so the Cloud API spec isn't versioned.
```

File: api-docs/README.md [hunk 9]
```diff
@@ -1,4 +1,172 @@
-# How to generate InfluxDB API docs
+# API reference documentation
+
+## How to update API docs for InfluxDB Cloud
```
Comment (kelseiv) [inline api-docs/README.md:3]:
```suggestion
## Update API docs for InfluxDB Cloud
```

File: api-docs/README.md [hunk 10]
```diff
@@ -1,4 +1,172 @@
...
+3. Enter the following commands into your terminal to generate the API docs with Redocly:
+
+   ```sh
+   sh generate-api-docs.sh
+   ```
+
+   To troubleshoot errors, see [how to generate API docs locally](#how-to-generate-api-docs-locally).
+4. Commit the changes, push your branch to `influxdata/docs-v2`, and create a PR against the `master` branch.
+
+## How to update API docs for a new InfluxDB OSS version release
```
Comment (kelseiv) [inline api-docs/README.md:33]:
```suggestion
## Update API docs for an InfluxDB OSS release
```

File: api-docs/README.md [hunk 11]
```diff
@@ -1,4 +1,172 @@
...
+9. Enter the following commands into your terminal to generate the API docs with Redocly:
+
+   ```sh
+   sh generate-api-docs.sh
+   ```
+
+   To troubleshoot errors, see [how to generate API docs locally](#how-to-generate-api-docs-locally).
+10. Commit the changes, push your branch to `influxdata/docs-v2`, and create a PR against the `master` branch.
+
+## How to update API docs for OSS spec changes between releases
```
Comment (kelseiv) [inline api-docs/README.md:111]:
```suggestion
## Update API docs for OSS spec changes between releases
```

**Review by kelseiv — APPROVED** [review]

File: api-docs/openapi/plugins/decorators/replace-shortcodes.js [hunk 12]
```diff
@@ -2,9 +2,12 @@ module.exports = ReplaceShortcodes;
 
 function replaceDocsUrl(field) {
   if(!field) { return }
-  const shortcode = '{{% INFLUXDB_DOCS_URL %}}';
+  /** Regex to match the URL "shortcode" {{% INFLUXDB_DOCS_URL %}}.
+   * [^]* matches line breaks.
+   */
+  const shortcodeRe = /\{\{[^]*%\s*[^]*INFLUXDB_DOCS_URL[^]*\s*[^]*%\}\}/g
```
Comment (sunbryely-influxdata) [inline api-docs/openapi/plugins/decorators/replace-shortcodes.js:8]: 🤯

**Review by sunbryely-influxdata — APPROVED** [review]

File: api-docs/openapi/plugins/decorators/replace-shortcodes.js [hunk 13]
```diff
@@ -2,9 +2,12 @@ module.exports = ReplaceShortcodes;
 
 function replaceDocsUrl(field) {
   if(!field) { return }
-  const shortcode = '{{% INFLUXDB_DOCS_URL %}}';
+  /** Regex to match the URL "shortcode" {{% INFLUXDB_DOCS_URL %}}.
+   * [^]* matches line breaks.
```
Comment (jstirnaman) [inline api-docs/openapi/plugins/decorators/replace-shortcodes.js:6]:
```suggestion
   * [^]* matches line breaks (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp#using_regular_expression_on_multiple_lines).
```

File: api-docs/README.md (hunk 6 above)
Comment (jstirnaman) [inline api-docs/README.md]:
```suggestion
3. Cherry-pick the commits with the updated description fields and push the commits to the remote branch, replacing **`[COMMIT_SHAs]`** (one or more commit SHAs (space-separated))--for example:

   ```sh
   git cherry-pick [COMMIT_SHAs]
   git push -f
   ```
```

File: api-docs/README.md [hunk 14]
```diff
@@ -1,4 +1,172 @@
...
+   ```sh
+   git push tags
+   ```
+
+5. Enter the following commands into your terminal to update `docs-release/influxdb-oss` branch to the OSS release commit, rebase or reset the branch to the [latest release of InfluxDB OSS](#how-to-find-the-api-spec-used-by-an-influxdb-oss-version).
+
+   ```sh
+   git checkout docs-release/influxdb-oss
+   git rebase -i OSS_RELEASE_TAG
+   git push -f origin docs-release/influxdb-oss
```
Comment (jstirnaman) [inline api-docs/README.md:61]:
```suggestion
5. Enter the following commands into your terminal to checkout a `docs-release/influxdb-oss` tracking branch, rebase the branch to the tagged release of InfluxDB OSS, and push the change to the`docs-release/influxdb-oss` remote branch:

   ```sh
   git checkout -t docs-release/influxdb-oss
   git rebase -i OSS_RELEASE_TAG
   git push -f
```

File: api-docs/v2.3/ref.yml [hunk 15]
```diff
@@ -9703,7 +9705,7 @@ paths:
         metrics](https://docs.influxdata.com/influxdb/v2.3/reference/internals/metrics/).
 
         - Learn how to use InfluxDB to [scrape Prometheus
-        metrics](https://docs.influxdata.com/influxdb/v2.3/write-data/developer-tools/scrape-prometheus-metrics/).
+        metrics](https://docs.influxdata.com/influxdb/v2.3write-data/developer-tools/scrape-prometheus-metrics/).
```
Comment (jstirnaman) [inline api-docs/v2.3/ref.yml:9708]:
```suggestion
        metrics](https://docs.influxdata.com/influxdb/v2.3/write-data/developer-tools/scrape-prometheus-metrics/).
```

File: api-docs/v2.3/swaggerV1Compat.yml [hunk 16]
```diff
@@ -9,7 +9,7 @@ info:
 
 
     If you want to use the latest InfluxDB /api/v2 API instead, see the
-    [InfluxDB v2 API documentation](/influxdb/v2.3/api/).
+    [InfluxDB v2 API documentation](/influxdb/v2.2/api/).
```
Comment (jstirnaman) [inline api-docs/v2.3/swaggerV1Compat.yml:12]:
```suggestion
    [InfluxDB v2 API documentation](/influxdb/latest/api/).
```

File: api-docs/openapi/plugins/decorators/replace-shortcodes.js (hunk 13 above)
Comment (jstirnaman) [inline api-docs/openapi/plugins/decorators/replace-shortcodes.js]:
```suggestion
   * [^]* matches line breaks. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp#using_regular_expression_on_multiple_lines
```

**Review by jstirnaman — COMMENTED** [review]
Link to MDN doc.

**Pull request merged**